src/main/scala/workflow/graph/DefaultOptimizer.scala

```diff
@@ -0,0 +1,11 @@
+package workflow.graph
+
+/**
+ * The default Pipeline optimizer used when executing pipelines.
+ */
+object DefaultOptimizer extends Optimizer {
+  protected val batches: Seq[Batch] =
+    Batch("Load Saved State", Once, ExtractSaveablePrefixes, SavedStateLoadRule, DanglingNodeRemovalRule) ::
```

Comment:
This seems fishy to me - but I'll come back to this after I review the execution logic.